lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+/* global document,window */
+
+/* istanbul ignore next */
+function collectTagsThatBlockFirstPaint() {
```

Comment:
`collectTagsThatBlockFirstPaint` has a lot of logic in it now, all which gets injected into the page. Would collecting the links/scripts on the page and doing the filtering in the gatherer be better? We'd be less likely to eat errors (although we have the try/catch), but it also makes things easier to debug/test. Code injected into the page is a black box.

cc @brendankenny wdyt, does it matter?